src/Arcus.EventGrid.Tests.Unit/Parsing/CloudOrEventGridEventParsingTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Linq;
+using Arcus.EventGrid.Contracts;
+using Arcus.EventGrid.Contracts.Interfaces;
+using Arcus.EventGrid.Parsers;
+using Arcus.EventGrid.Tests.Core.Events;
+using Arcus.EventGrid.Tests.Unit.Artifacts;
+using CloudNative.CloudEvents;
+using Microsoft.Azure.EventGrid.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Arcus.EventGrid.Tests.Unit.Parsing
+{
+    public class CloudOrEventGridEventParsingTests
```

Comment:
Can we extend these some more, for example:
- Input as EventGrid, Use AsCloudEvent (and otherwise)

Reply:
Yep

Reply:
Was done I think.

Reply:
Yep, done.